enhancements/olm/simplify-apis.md

```diff
@@ -0,0 +1,523 @@
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109719
+    - kind: ClusterServiceVersion
+      namespace: operators
+      name: plumbus.v2.0.0-alpha
+      uid: d70a53b5-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: operators.coreos.com/v1alpha1
+      resourceVersion: 109811
+      conditions:
+      - type: Installing
+        status: True
+        reason: AllPreconditionsMet
+        message: deployment rolling out
+        lastTransitionTime: "2019-09-16T22:26:29Z"
+    - kind: CustomResourceDefinition
+      name: plumbai.how.dotheydoit.com
+      uid: d680c7f9-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: apiextensions.k8s.io/v1beta1
+      resourceVersion: 75396
+    - kind: ClusterRoleBinding
+      namespace: operators
+      name: rb-9oacj
+      uid: d81c24d6-d06a-11e9-821f-9a2e3b9d8156
+      apiVersion: rbac.authorization.k8s.io/v1
+      resourceVersion: 75438
+```
+
+### User Stories
+
+#### As a OLM Admin, I want to
+
+- restrict the resources OLM will apply from a bundle image/index when installing an operator
+- restrict the ServiceAccounts a user can install operators with
+- restrict the bundle indexes a given user can use to resolve operator bundles
+
+#### As an Operator Installer, I want to
+
+- view the state of an operator by inspecting the status of a single API resource
+- deploy an operator by applying its manifests directly
+- deploy an operator using a bundle image
+- deploy and upgrade an operator by referencing a bundle index
+
+#### As an Operator User, I want to
+
+- view the operators I can use on a cluster
+
+### Implementation Details/Notes/Constraints
+
+#### The `Operator` Resource
+
+The `operators.coreos.com/v2alpha1` API group version will be added to OLM
+
+- alpha versions make no [support guarantees](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions), so resources can be iterated quickly
+
+as well as a command option on `olm-operator` that enables `v2alpha1`
+
+- leave disabled by default
+
+```bash
+olm-operator --enable-v2alpha1
+```
+
+The cluster-scoped resource, `Operator` will be added to `v2alpha1`
+
+- cluster scoped resources can be listed without granting namespace list, but do require a `ClusterRole`
+- namespace scoped resources can specify references to cluster scoped resources in their owner references, which lets us use garbage collection to clean up resources across mutliple namespaces; e.g. `RoleBindings` copied for an `OperatorGroup`
+
+#### Constraining Installation
+
+In a fashion similar `OperatorGroups`, we can constrain the installation of an operator by associating it with a `ServiceAccount`. For an `Operator` resource, the user's desired `ServiceAccount` will be specified by the `spec.install.as` field. The install namespace, where the namespaced contents of an operator will be installed to, is specified by the `install.namespace` field.
```

Comment:
What is driving this use case? Most admins are used to the idea of running a package manager as "root" as long as they can:

1. have an opportunity to see what it's going to do (in this case, just list the resources that will be created)
2. are able to limit the sources from which content is installed to sources they trust

The far more interesting concern is what permissions the operator itself runs with over the long term. That is a better fit for your android update analogy below. It's also observable from the operator bundle, so it's mostly a solved problem (unless you want to influence auto-upgrade gating based on RBAC changes).

When installing an operator, won't that in most cases involve creating a ServiceAccount, ClusterRole, and ClusterRoleBinding for that operator? If the installing user has the ability to create those things, don't they already have the keys to the kingdom? What value will we get from tracking resource-by-resource what that user can do vs what it takes to install the operator? Would it be sufficient to just assume that installing operators takes a lot of privilege, and gate that with RBAC for the `Operator` resource?

Reply:
The user or SA creating a ClusterRoleBinding cannot do so unless they already have the permissions being granted.  For managed services (OSD, ARO) a customer does not run as a cluster-admin.  In this case we cannot allow customers to install any arbitrary operator as they could elevate privileges.

Reply:
What does ARO stand for?

Reply:
> What is driving this use case? Most admins are used to the idea of running a package manager as "root"...

As @jewzaam points out, the admin installing an operator is not always a full blown cluster-admin. 

> The far more interesting concern is what permissions the operator itself runs with over the long term... It's also observable from the operator bundle, so it's mostly a solved problem (unless you want to influence auto-upgrade gating based on RBAC changes).

We plan on doing this. The recent proposal updates (hopefully) make this more clear.

> If the installing user has the ability to create those things, don't they already have the keys to the kingdom?

(kube's RBAC model)[https://kubernetes.io/docs/reference/access-authn-authz/rbac/#privilege-escalation-prevention-and-bootstrapping] prevents escalation by _just_ having permission to create/update (Cluster)Roles/RoleBindings.

> What value will we get from tracking resource-by-resource what that user can do vs what it takes to install the operator?

We prevent users from escalating their privilege via OLM.

Reply:
> What does ARO stand for?

[Azure Red Hat OpenShift](https://azure.microsoft.com/en-us/services/openshift/)